### PR TITLE
Remove FP16 specific models from imagenet example

### PR DESCRIPTION
**Pull request by takagi**
This PR removes FP16 specific models from imagenet example as we now have the default dtype feature, which perfectly works to run the example in FP16 mode without any change of the script.

### CODE REVIEW

**Comment by takagi** [comment]
Remaining is to check if it works properly with `--dali` option.

**Comment by takagi** [comment]
Removed `AlexFp16` and `GoogLeNetBNFp16` uses `HeNormal(1 / np.sqrt(2))` as  the initializer of their weight parameters, which is equivalent to `LeCunNormal()` that is used in `AlexFp16` and `GoogleNetBN`.

File: examples/imagenet/README.md [hunk 1]
```diff
@@ -17,4 +17,15 @@ It requires the training and validation dataset of following format:
...
 
-This example requires "mean file" which is computed by `compute_mean.py`.
+This example requires "mean file" which is computed by `compute_mean.py`:
+
+```
+python compute_mean.py <path to training list>
+python train_imagenet.py <path to training list> <path to validation list>
+```
+
+With `CHAINER_DTYPE` environment variable, you can run the example with FP16 mode:
```
[the remark this answers is not in the document]
Comment (takagi) [reply examples/imagenet/README.md]: Okay, I will change it to use a command line argument!

File: examples/imagenet/README.md [hunk 2]
```diff
@@ -17,4 +17,15 @@ It requires the training and validation dataset of following format:
...
 
-This example requires "mean file" which is computed by `compute_mean.py`.
+This example requires a "mean file," which is computed by `compute_mean.py`:
+
+```
+python compute_mean.py <path to training list>
+python train_imagenet.py <path to training list> <path to validation list>
+```
+
+With the `CHAINER_DTYPE` environment variable, you can run the example in FP16 mode:
```
Comment (Crissman) [inline examples/imagenet/README.md]:
```suggestion
With the `--dtype` argument, you can run the example in FP16 mode:
```

File: examples/imagenet/README.md [hunk 3]
```diff
@@ -17,4 +17,15 @@ It requires the training and validation dataset of following format:
...
+
+```
+CHAINER_DTYPE=float16 python train_imagenet.py <path to training list> <path to validation list>
```
Comment (Crissman) [inline examples/imagenet/README.md]:
```suggestion
python train_imagenet.py --dtype float16 <path to training list> <path to validation list>
```

**Comment by takagi** [comment]
I applied the changes!

**Comment by Crissman** [comment]
Jenkins, test this please

**Comment by Crissman** [comment]
Jenkins test needs to be rerun. Jenkins, test this please.

**Comment by Crissman** [comment]
Appveyor failure looks unrelated. Merging!

**Comment by kmaehashi** [comment]
@Crissman Could you make a backport PR?

**Comment by kmaehashi** [comment]
I made a backport.

**Pull request merged**